src/zenml/logger.py

```diff
@@ -77,7 +78,11 @@ def format(self, record: logging.LogRecord) -> str:
         if ZENML_LOGGING_COLORS_DISABLED:
             # If color formatting is disabled, use the default format without colors
             formatter = logging.Formatter(self.format_template)
-            return formatter.format(record)
+            formatted_message = formatter.format(record)
+            # Add warning icon for warning messages even when colors are disabled
+            if record.levelno == LoggingLevels.WARN.value:
+                formatted_message = "⚠ " + formatted_message
+            return formatted_message
```

Comment:
```suggestion
            return formatter.format(record)
```

Similar to the checkmark, I would not alter the message with a warning sign here.